### PR TITLE
Fixed #31645 --  Enhanced the migration warning for migrate commmand.

### PR DESCRIPTION
**Pull request by chinmoy12c**
Ticket: https://code.djangoproject.com/ticket/31645

### CODE REVIEW

**Comment by chinmoy12c** [comment]
Ticket : [https://code.djangoproject.com/ticket/31645](url)

File: django/core/management/commands/migrate.py [hunk 1]
```diff
@@ -235,6 +235,15 @@ def handle(self, *args, **options):
...
                     ))
+                    for app_label, app_migrations in changes.items():
+                        self.stdout.write(self.style.WARNING(
+                            " Migrations for \'{}\':".format(app_label)
+                        ))
+                        for migration in app_migrations:
+                            for operation in migration.operations:
+                                self.stdout.write(self.style.WARNING(
+                                    "  - {}".format(operation.describe())
+                                ))
```
Comment (felixxm) [inline django/core/management/commands/migrate.py:246]: IMO we don't need a list of all operations in this message, it's too detailed. If someone would like to check the list of operations they can always use `makemigrations --dry-run`.

Adding a list of apps should be enough, e.g.
```diff
diff --git a/django/core/management/commands/migrate.py b/django/core/management/commands/migrate.py
index 81696f11c1..32b1462319 100644
--- a/django/core/management/commands/migrate.py
+++ b/django/core/management/commands/migrate.py
@@ -227,8 +227,9 @@ class Command(BaseCommand):
                 changes = autodetector.changes(graph=executor.loader.graph)
                 if changes:
                     self.stdout.write(self.style.NOTICE(
-                        "  Your models have changes that are not yet reflected "
-                        "in a migration, and so won't be applied."
+                        "  Your models in apps [%s] have changes that are not "
+                        "yet reflected in a migration, and so won't be "
+                        "applied." % ', '.join(changes)
                     ))
                     self.stdout.write(self.style.NOTICE(
                         "  Run 'manage.py makemigrations' to make new "
```

Comment (chinmoy12c) [reply django/core/management/commands/migrate.py:246]: Thank you for the review @felixxm I'll make the adjustments right away :)

Comment (chinmoy12c) [reply django/core/management/commands/migrate.py]: Fixed the code to only list app names. Also @felixxm could you please elaborate on the kind of tests necessary for this?

Comment (felixxm) [reply django/core/management/commands/migrate.py]: We should be able to assert `stdout`. You can find similar tests in `tests/migrations/test_commands.py`.

Comment (chinmoy12c) [reply django/core/management/commands/migrate.py]: I've added the tests please review it :)

**Comment by felixxm** [comment]
@chinmoy12c Thanks :+1: Welcome aboard :boat: 

I simplified tests.

**Comment by chinmoy12c** [comment]
> @chinmoy12c Thanks  Welcome aboard 
> 
> I simplified tests.

Thanks :)

File: django/core/management/commands/migrate.py [hunk 2]
```diff
@@ -227,8 +227,9 @@ def handle(self, *args, **options):
                 changes = autodetector.changes(graph=executor.loader.graph)
                 if changes:
                     self.stdout.write(self.style.NOTICE(
-                        "  Your models have changes that are not yet reflected "
-                        "in a migration, and so won't be applied."
+                        "  Your models in app(s): %s have changes that are not "
+                        "yet reflected in a migration, and so won't be "
+                        "applied." % ", ".join(repr(app) for app in changes)
```
Comment (kezabelle) [inline django/core/management/commands/migrate.py:232]: Do these render in alphabetical order for consistency with `63a6a653d4282f74e35582d41ca507139c9cfbe1`?

Comment (felixxm) [reply django/core/management/commands/migrate.py:232]: Good point, fixed.

**Comment by felixxm** [comment]
Merged in 2928019e0ccd8e9c7d3a3fba7722a7af87018e5d.

**Comment by jayvdb** [comment]
Thanks @chinmoy12c

**Pull request closed without merging**